api/types/types.proto

```diff
@@ -827,6 +827,9 @@ message ProvisionTokenSpecV2 {
         [ (gogoproto.jsontag) = "join_method", (gogoproto.casttype) = "JoinMethod" ];
     // BotName is the name of the bot this token grants access to, if any
     string BotName = 5 [ (gogoproto.jsontag) = "bot_name,omitempty" ];
+    // RefResourceID is an identifier to correlate this token to resources that used it to join the
+    // cluster
+    string RefResourceID = 6 [ (gogoproto.jsontag) = "ref_resource_id,omitempty" ];
```

Comment:
Can we have a generic `suggested_labels` map field instead? It could be useful in the future, maybe to add labels from the web UI, to be embedded in the script ("suggested" or "initial" because they're not enforced and can be changed from the node's config file, as opposed to some hypothetical centrally managed and enforced labeling on the cluster side).

Reply:
Something like this?
https://github.com/gravitational/teleport/pull/15114/commits/bebe1426c7d7499558d5dbde5e8880776f24dd02

Can you please take a look?

Reply:
I wonder if we should do this in the backend or let the UI/client set the labels freely :thinking: 

IMO, I think we should do this in the backend as opposed to let the UI generate the ID and set the label
However, given the current flow I start to doubt if that's the correct approach